astro/src/content/blog/custom-scopes-in-third-party-applications.mdx

```diff
@@ -0,0 +1,96 @@
+
+How is this implemented?
+
+## FusionAuth Setup For Third Party Application
+Change Bank is lucky because they use FusionAuth and it supports custom scopes as of version 1.50.0.
+
+The first thing they will need to do is create the Budget Buddy application in their FusionAuth instance. Notice that the Budget Buddy Client Id starts with `4581` and the Change Bank Client Id starts with `e9fd`. Change Bank will need to provide the Budget Buddy organization with the Client Id that begins with `4581`, the Client Secret for the Budget Buddy application, and the Change Bank login url. 
+
+![Adding Budget Buddy Application.](/img/blogs/custom-scopes-in-third-party-applications/new-budget-buddy-application.png)
+
+Then, they will need to configure the application as a third party application by updating the `Scopes` settings for the application. They should set the Relationship to `Third-party. The application is external to the authorization server.`
+
+![Budget Buddy Scopes Settings.](/img/blogs/custom-scopes-in-third-party-applications/budget-buddy-scopes.png)
+
+Next, they will need to add the custom scope for the Budget Buddy application. In this case, the custom scope `balance:read` has been created. This will be the scope that the Change Bank API will look for when determining if the application can retrieve the balance.
+
+![Budget Buddy Custom Scopes.](/img/blogs/custom-scopes-in-third-party-applications/budget-buddy-custom-scopes.png)
+
+## Updating Change Bank API
+
+Now all that is left is to create the `get-balance' endpoint for the API. To accomplish this, a new controller (along with a bit of internal plumbing) is added to the API project. 
+
+<Aside type="note">
+The APIs use the [rack-jwt](https://github.com/FusionAuth/rack-jwt) Gem to help with parsing and validation of the token that will be sent with the API request. The 'internal plumbing' of the API is beyond the scope (no pun intended) of this blog. For more details, please see the [FusionAuth API Quickstart for Ruby on Rails](https://github.com/FusionAuth/fusionauth-quickstart-ruby-on-rails-api). 
+</Aside>
+
+The important part for the code in the controller is that it checks for the correct Budget Buddy Client ID and checks for the scope `balance:read`.  If those values are confirmed, then a balance is returned. If this were a real application, it would also retrieve the `change_bank_user_id` and look up the actual balance for the user.
```

Comment:
```suggestion
The important part of the code in the controller is that it checks for the correct Budget Buddy Client Id and checks for the scope `balance:read`.  If those values are confirmed, then a balance is returned. If this were a real application, it would also retrieve the `change_bank_user_id` and look up the actual balance for the user.
```